grails-app/domain/org/pih/warehouse/core/User.groovy

```diff
@@ -84,6 +84,11 @@ class User extends Person {
         return defaultRoles
     }
 
+    boolean hasPrimaryRole(Location currentLocation) {
+        def roles = getEffectiveRoles(currentLocation)
+        return roles.roleType.find { RoleType.listPrimaryRoleTypes().contains(it) } ? true : false
```

Comment:
Not to be nitpicky, but the `? true : false` is redundant so you can remove it. The find method will return the first object it finds that matches the closure logic (true) or it will return null (false). I usually wouldn't correct this since the explicit approach usually increases readability. However, in this case I think it makes it slightly more confusing.